Detectors/EMCAL/simulation/src/Digitizer.cxx

```diff
@@ -0,0 +1,128 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "EMCALSimulation/Digitizer.h"
+#include "EMCALBase/Digit.h"
+#include "EMCALBase/Geometry.h"
+#include "EMCALBase/GeometryBase.h"
+#include "EMCALBase/Hit.h"
+#include "MathUtils/Cartesian3D.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+
+#include <TRandom.h>
+#include <TVector3.h>
+#include <climits>
+#include <forward_list>
+#include "FairLogger.h" // for LOG
+
+ClassImp(o2::EMCAL::Digitizer);
+
+// using o2::EMCAL::GeometryBase;
+using o2::EMCAL::Digit;
+using o2::EMCAL::Hit;
+
+using namespace o2::EMCAL;
+
+//_______________________________________________________________________
+void Digitizer::init() {}
+
+//_______________________________________________________________________
+void Digitizer::finish() {}
+
+//_______________________________________________________________________
+void Digitizer::process(const std::vector<Hit>* hits, std::vector<Digit>* digits)
+{
+  digits->clear();
+  mDigits.clear();
+
+  for (auto hit : *hits) {
```

Comment:
if your intention is just to look at hits then: `for (const auto& hit: *hits) {`
also making `HitToDigit(const Hit&)` (as per comment of @sawenzel ) will improve both the contract and performance (in general).